frontend/src/controllers/LoginState.js

```diff
@@ -19,6 +19,9 @@ export default function updateLoginState(attempts) {
     .get(url, { withCredentials: true })
     .then(result => {
       store.commit("setUsername", result.data.username);
+      if(typeof callback === 'function') {
+        callback();
+      }
```

Comment:
On line 32 (won't let me inline comment it), we want to pass `callback` to `updateLoginState` otherwise we'll drop the callback if there's an error on the first attempt, but a subsequent attempt succeeds.

Reply:
Added callback param to the retry call